sys/net/gnrc/network_layer/sixlowpan/frag/sfr/gnrc_sixlowpan_frag_sfr.c

```diff
@@ -0,0 +1,1546 @@
+                    DEBUG("         (next ARQ timeout in %lu)\n",
+                          (long unsigned)next_arq_offset);
+                }
+                reschedule_arq_timeout = true;
+            }
+            else if (_frag_ack_req(frag_desc)) {
+                if ((frag_desc->retries++) < CONFIG_GNRC_SIXLOWPAN_SFR_FRAG_RETRIES) {
+                    DEBUG("6lo sfr: %u retries left for fragment (tag: %u, "
+                          "X: %i, seq: %u, frag_size: %u, offset: %u)\n",
+                          CONFIG_GNRC_SIXLOWPAN_SFR_FRAG_RETRIES -
+                          (frag_desc->retries - 1), (uint8_t)fbuf->tag,
+                          _frag_ack_req(frag_desc), _frag_seq(frag_desc),
+                          _frag_size(frag_desc), frag_desc->offset);
+                    if (_resend_frag(&frag_desc->super, fbuf) != 0) {
+                        goto error;
+                    }
+                    reschedule_arq_timeout = true;
+                }
+                else {
+                    DEBUG("6lo sfr: resending fragment (tag: %u, X: %i, seq: %u, "
+                          "frag_size: %u, offset: %u)\n",
+                          (uint8_t)fbuf->tag, _frag_ack_req(frag_desc),
+                          _frag_seq(frag_desc), _frag_size(frag_desc),
+                          frag_desc->offset);
+                    _retry_datagram(fbuf);
+                    return;
+                }
+            }
+            else {
+                DEBUG("6lo sfr: nothing to do for fragment %u\n",
+                      _frag_seq(frag_desc));
+            }
+        } while (frag_desc != head);
+    }
+    assert(fbuf->sfr.frags_sent == clist_count(&fbuf->sfr.window));
+    if (reschedule_arq_timeout) {
+        _sched_arq_timeout(fbuf, next_arq_offset);
+        return;
+    }
+error:
+    /* don't check return value, as we don't want to wait for an ACK again ;-) */
+    _send_abort_frag(fbuf->pkt, (uint8_t)fbuf->tag, false, 0);
+    _clean_up_fbuf(fbuf);
+}
+
+void gnrc_sixlowpan_frag_sfr_inter_frame_gap(void)
+{
+    if (CONFIG_GNRC_SIXLOWPAN_SFR_INTER_FRAME_GAP_US > 0) {
+        _frame_queue_t *node = (_frame_queue_t *)clist_lpop(&_frame_queue);
+
+        if (node != NULL) {
+            _last_frame_sent = xtimer_now_usec();
+            gnrc_sixlowpan_dispatch_send(node->frame, NULL, node->page);
+            /* unset packet just to be safe */
+            node->frame = NULL;
+            clist_rpush(&_frame_queue_free, &node->super);
+        }
+        if (clist_lpeek(&_frame_queue) != NULL) {
+            _sched_next_frame();
+        }
+    }
+}
+
+/* ====== INTERNAL FUNCTION DEFINITIONS ====== */
+static inline uint16_t _min(uint16_t a, size_t b)
+{
+    return (a < b) ? a : (uint16_t)b;
+}
+
+static inline kernel_pid_t _getpid(void)
+{
+    return IS_ACTIVE(TEST_SUITES) ? gnrc_sixlowpan_get_pid() : thread_getpid();
+}
+
+static void _clean_slate_datagram(gnrc_sixlowpan_frag_fb_t *fbuf)
```

Comment:
So this resets the datagram to the state before sending?

Reply:
Before fragmenting. SFR allows for resending the whole datagram if fragmentation timed out.

Reply:
I added some doc to explain what this function does and when it can be used.